tools/m1_utils/update_static.sh

```diff
@@ -10,7 +10,21 @@ unarchive() {
 
     rm -rf *.o 
     ar -x "$FILE" 
-    ar t "$FILE" | grep \.o$ | sort --ignore-case | uniq -D -i > duplicate_objs.txt
+    ar t "$FILE" | grep \.o$ | sort --ignore-case > objs.txt
+
+    # What this awk command does:
+    # ------------------------
+    # count = {} # define empty dictionary count
+    # if NR == FNR: #if first time iterating through objs.txt
+    #     for line in the objs.txt: 
+    #         count[lowercase(line)] += 1  # count the number of times each line showed up in the file (case insensitive)
+    #         continue 
+    # else: 
+    #     for line in the objs.txt: 
+    #         if count[lowercase(line)] > 1: 
+    #             print(line, dest=duplicate_objs.txt) # print only when this has appear more than once in objs.txt (case insensitive)
+    # ------------------------
+    awk 'NR==FNR {count[tolower($0)]++; next} count[tolower($0)]>1' objs.txt objs.txt > duplicate_objs.txt
```

Comment:
This looks fine and is similar what the original code did: it originally relied on the sorted property to loop them one time; isn't the bulk of the perf issue though.

If performance is a problem for you, we can consider using Tapans impl. If it maintains and parity with unarchiving edge cases. We can also build one from the original code: e.g. calling code from `cctools` similar to how [`print`](https://github.com/apple-oss-distributions/cctools/blob/main/ar/print.c#L104) in `ar` does: pulling `cctools` as a `http_archive` - compiling a `cc_library` then call `ar_getobj` our own unarchiver. Finally, the other idea is to handle archives in `arm64-to-sim`. This is a lot for something that caches though